### ISSUE TITLE
Thumbnail URLs require you to be signed in to OneDrive(?)

### ISSUE THREAD

**Issue by Mike-Bell** [microsoftgraph/microsoft-graph-docs#135]
I'm able to get thumbnail URL's as per http://graph.microsoft.io/en-us/docs/api-reference/v1.0/api/thumbnailset_get. A logical thing I might want to do is to plop the `url` value from this response into an image tag on a HTML page. Unfortunately, this only works if the user is already signed in directly to OneDrive. If the user does not have an active login session with OneDrive (which is a legitimate possibility if I'm developing an app that uses OneDrive as backing storage), then the browser is unable to retrieve the image contents and instead returns a 403.

Is there some method for fetching pre-authenticated thumbnail URLs? I tried grabbing an access token for the OneDrive API and appending that to the URL (e.g. `preAuthUrl = url + '?access_token=' + token`), but this still 403s (I thought this method of passing an access token was supported in the OneDrive API?). I see that I can retrieve the thumbnail contents via xhr if I use my access token in the header. I guess I could try constructing a data URL from the encoded contents and putting that in my tag, but I'd like to avoid this.

TLDR wondering if there's some way to get or construct a pre-authenticated url for thumbnails. Something like an equivalent of `@content.downloadUrl / @microsoft.graph.downloadUrl` but for thumbs.

**Comment by marych** [comment]
We're working on returning pre-authenticated thumbnail URLs and expect it to be available within about a month.

**Comment by Mike-Bell** [comment]
Thanks for the response @marych !

**Comment by Mike-Bell** [comment]
For anyone wondering, I found a temporary work-around. I can fetch the image data via xhr using an access token for the onedrive API. If I set `responseType: 'blob'` when I do so, and then use FileReader.readAsDataURL to get a data URL that I can plop into an image tag.

**Comment by rgregg** [comment]
Thanks for posting your workaround. This sounds like the best solution until we ship pre-authenticated thumbnail URLs.

**Comment by michaelva** [comment]
Running into the same issue. @marych Any update on the pre-authenticated thumbnail URL?

**Comment by marych** [comment]
The change was implemented and enabled as planned, but we needed to disable it as the change in behavior broke some clients. We expect to enable it again in about a month.

**Comment by raybaranoski** [comment]
Hi, is this specific to OneDrive or does this also apply to OneDrive For Business?  I am having difficulty getting thumbnails via the Graph api for OneDrive For Business.  I can get the urls fine, but I can't get the urls to load and you can't just put the url in an img tag.  I posted the details at [](http://stackoverflow.com/q/38065469/6512942)...  Any help would be appreciated...  Thanks

**Comment by marych** [comment]
It applies to OneDrive for Business. The rollout of the fix has been delayed/blocked by a consuming application relying on the current URL format.

**Comment by rgregg** [comment]
Good news folks: we're rolling out the fix to this to our Microsoft tenant starting today, and assuming everything goes well, we'll start rolling the fix out to everyone next week. The existing issue that was blocking this fix has been mitigated and we should have this problem resolved very soon.

**Comment by mattgeim** [comment]
Feel free to re-open this issue if needed - it appears it has been resolved per @rgregg.  Thanks!

**Comment by dcakirkaya** [comment]
2020 and One Drive for business thumbnail URLs do expire and I could find nothing in the documentation about it except this issue.
